File: app/src/main/java/com/kimjisub/launchpad/api/file/FileApi.java
package com.kimjisub.launchpad.api.file;

import com.kimjisub.launchpad.api.BaseApiService;
import com.kimjisub.launchpad.manager.Log;

import okhttp3.OkHttpClient;
import okhttp3.Response;
import okhttp3.ResponseBody;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Streaming;

public class FileApi extends BaseApiService {

	static final String APIURL = "https://api.unipad.kr";
	static FileService fileService;

	public static FileService getService() {
		if (fileService == null) {
			HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor(Log::network)
					.setLevel(HttpLoggingInterceptor.Level.BODY);
			OkHttpClient client = getUnsafeOkHttpClient()
					.addInterceptor(interceptor)
					//.cookieJar(new MyCookieJar())
					.build();

			Retrofit retrofit = new Retrofit.Builder()
					.baseUrl(APIURL)
					.addConverterFactory(GsonConverterFactory.create(getGson()))
					.client(client)
					.build();
			fileService = retrofit.create(FileService.class);
		}

		return fileService;
	}



	public interface FileService {

		// ============================================================================================= /unishare

		@GET("/unishare/{code}/download")
		@Streaming
		Call<ResponseBody> unishare_download(@Path("code") String code);
	}
}

File: app/src/main/java/com/kimjisub/launchpad/api/file/FileApi.kt
package com.kimjisub.launchpad.api.file

import com.kimjisub.launchpad.api.BaseApiService
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Streaming
import retrofit2.http.Url

object FileApi {
	private const val API_BASE_URL = "https://api.unipad.io"

	val service: FileService by lazy {
		BaseApiService.createRetrofitService(API_BASE_URL, FileService::class.java)
	}

	interface FileService {
		@GET
		@Streaming
		fun download(@Url url: String): Call<ResponseBody>
	}
}
